lib/matplotlib/backends/backend_svg.py

```diff
@@ -370,23 +371,30 @@ def _write_metadata(self, metadata):
             else:
                 metadata['Date'] = datetime.datetime.today().isoformat()
 
-        mid = writer.start('metadata')
-        writer.start('rdf:RDF', attrib={
+        mid = None
+        def ensure_metadata():
+            nonlocal mid
```

Comment:
since you use `mid` externally anyway, can you not just pass this to `ensure_metatada` and return the new "mid" state?  I think we very rarely use `nonlocal` objects to keep scope clear?

Reply:
Sure.... I feel like requiring the user to explicitly maintain the state isn't as clean since the whole point was to automatically maintain the state for the user. If this is the preferred way I can certainly do it though.

Reply:
I don't mean the user should maintain it, just that the helper takes mid as an argument and returns it rather than rely on "nonlocal".

Reply:
By "user" I mean the caller of `ensure_metadata`. I've already updated the patch...

Reply:
OK, thanks.  *I* prefer that, but I don't claim any great expertise in what is most pythonic so other reviewers should also chime in.